source-batchsql/main.go

```diff
@@ -0,0 +1,480 @@
+package main
+
+import (
+	"context"
+	"database/sql"
+	"encoding/json"
+	"fmt"
+	"net/url"
+	"reflect"
+	"regexp"
+	"strings"
+	"text/template"
+	"time"
+
+	schemagen "github.com/estuary/connectors/go/schema-gen"
+	boilerplate "github.com/estuary/connectors/source-boilerplate"
+	pc "github.com/estuary/flow/go/protocols/capture"
+	pf "github.com/estuary/flow/go/protocols/flow"
+	"github.com/invopop/jsonschema"
+	log "github.com/sirupsen/logrus"
+	"golang.org/x/sync/errgroup"
+
+	_ "github.com/jackc/pgx/v4/stdlib"
+)
+
+// Config tells the connector how to connect to and interact with the source database.
+type Config struct {
```

Comment:
sanity checking - you believe `Config` is extensible to other DBs ? How might that work?